interface/ceed-fortran.c

```diff
@@ -763,12 +860,18 @@ void fCeedQFunctionAddOutput(int *qf, const char *fieldname,
 
 #define fCeedQFunctionSetContext \
     FORTRAN_NAME(ceedqfunctionsetcontext,CEEDQFUNCTIONSETCONTEXT)
-void fCeedQFunctionSetContext(int *qf, CeedScalar *ctx, CeedInt *n, int *err) {
+void fCeedQFunctionSetContext(int *qf, int *ctx, int *err) {
   CeedQFunction qf_ = CeedQFunction_dict[*qf];
+  CeedUserContext ctx_ = CeedUserContext_dict[*ctx];
 
-  fContext *fctx = qf_->ctx;
-  fctx->innerctx = ctx;
-  fctx->innerctxsize = ((size_t) *n)*sizeof(CeedScalar);
+  CeedUserContext fctx;
+  *err = CeedQFunctionGetContext(qf_, &fctx);
+  if (*err) return;
+  CeedFortranContext fctxdata;
+  *err = CeedUserContextGetData(fctx, CEED_MEM_HOST, (void **)&fctxdata);
+  if (*err) return;
+  fctxdata->innerctx = ctx_;
+  *err = CeedUserContextRestoreData(fctx, (void **)&fctxdata);
```

Comment:
I'm still not super happy about Fortran hijacking the context to make the interface work, but I couldn't think of a better way to do this given the limitations of Fortran77